mantidproject/mantid
# Muon - Correct Workspace Naming in New Algorithms

Closes #22623 , merge after #22619 is merged.

As per the linked issue, the naming conventions of the new muon algorithms `ApplyMuonDetectorGrouping` and `ApplyMuonDetectorGroupPairing` do not match the Muon Analysis GUI. This PR fixes that issue.

**Test 1**

- Open the Muon Aanlysis GUI (Interfaces->Muon->Muon Analysis), select the MUSR instrument. In the load run box give *15189* and hit enter. Data should appear in the ADS. Close the GUI.

- Open the `ApplyMuonDetectorGrouping` algorithm (in `Muon/Data Handling/`). Set the 

	- **InputWorkspace** : "MuonAnalysis"
	- **InputWorkspaceGroup** : "MUSR00015189" 
	- **GroupName** : "test"
	- **Grouping** : 1,2
	- **AnalysisType** : test both options (leaving **TimeMin** and **TimeMax** as defaults)
	- **RebinArgs** : leave blank
	- **TimeOffset** : 0.0

- Now run the algorithm, check that named workspaces are added to the workspace group (for the naming convention see the algorithm documentation). 

- Now change **SummedPeriods** to "1,2", check that the names are updated to include "; 1+2;". Now change the **SubtractedPeriods** to "2" and **SummedPeriods** to "1", the name should include "; 1-2;".

**Test 2**

- Follow the same instructions as above for the `ApplyMuonDetectorGroupPairing` algorithm, with the additional properties:

	- **Alpha** : 1.0
	- **SpecifyGroupsManually** : checked
	- **InputWorkspace** : MuonAnalysis
	- **Group1** and **Group2** : "1-5", "6-10"

Fixes #22623 

Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
